scipy/spatial/transform/_rigidtransformation.pyx

```diff
@@ -0,0 +1,1899 @@
+                    if dets[ind] <= 0:
+                        raise ValueError("Non-positive determinant in rotation component of "
+                                         f"matrix {ind}: {matrix[ind]}.")
+                    else:
+                        # Orthonormalize the rotation matrix
+                        U, _, Vt = np.linalg.svd(matrix[ind, :3, :3])
+                        matrix[ind, :3, :3] = U @ Vt
+                else:
+                    raise ValueError("Expected rotation component of transformation "
+                                     f"matrix {ind} be orthonormal: {matrix[ind]}.")
+
+        self._matrix = matrix
+
+    @cython.embedsignature(True)
+    @classmethod
+    def from_rotation(cls, rotation):
+        """Initialize from a rotation, without a translation.
+
+        When applying this transformation to a vector, the result is the same
+        as if the rotation was applied to the vector.
+        ``t.from_rotation(r).apply(vector) == r.apply(vector)``
+
+        Parameters
+        ----------
+        rotation : `Rotation` instance
+            A single rotation or a stack of rotations.
+
+        Returns
+        -------
+        transformation : `RigidTransformation` instance
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import RigidTransformation as T
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> import numpy as np
+
+        Creating a transformation from a single rotation:
+
+        >>> r = R.from_euler("ZYX", [90, 30, 0], degrees=True)
+        >>> r.apply([1, 0, 0])
+        array([0.       , 0.8660254, -0.5     ])
+        >>> t = T.from_rotation(r)
+        >>> t.apply([1, 0, 0])
+        array([0.       , 0.8660254, -0.5     ])
+        >>> t.single
+        True
+
+        The upper 3x3 submatrix of the transformation matrix is the rotation
+        matrix:
+
+        >>> np.allclose(t.as_matrix()[:3, :3], r.as_matrix(), atol=1e-12)
+        True
+
+        Creating multiple transformations from a stack of rotations:
+
+        >>> r = R.from_euler("ZYX", [[90, 30, 0], [45, 30, 60]], degrees=True)
+        >>> r.apply([1, 0, 0])
+        array([[0.        , 0.8660254 , -0.5       ],
+               [0.61237244, 0.61237244, -0.5       ]])
+        >>> t = T.from_rotation(r)
+        >>> t.apply([1, 0, 0])
+        array([[0.        , 0.8660254 , -0.5       ],
+               [0.61237244, 0.61237244, -0.5       ]])
+        >>> t.single
+        False
+        >>> len(t)
+        2
+        """
+        rotmat = rotation.as_matrix()
+        if rotation.single:
+            rotmat = rotmat[None, :, :]
+        num_transformations = len(rotmat)
+        matrix = np.zeros((num_transformations, 4, 4), dtype=float)
+        matrix[:, :3, :3] = rotmat
+        matrix[:, 3, 3] = 1
+        if rotation.single:
+            matrix = matrix[0]
+        return cls(matrix, normalize=False, copy=False)
+
+    def __repr__(self):
```

Comment:
Put the method in the other place?

Reply:
Where?

Reply:
Maybe right after `__init__`? Or somewhere at the end, where `__len__` and other special methods are located.